### DISCUSSION TITLE
Reading command parameters

### DISCUSSION THREAD

**Discussion by tassoman** [telegraf/telegraf#1291] [Q&A] [answered]
How can I listen for command having parameters? For example, user said `/auth username password` 🤔 
Looks like a silly question but I feel hard for newcomers reading documentation without any code example.

**Comment by wojpawlik** [comment, marked as answer]
Great question! Possibly related to #898.

Telegraf 4.0.0 doesn't include unopinionated command arguments parser, and opinionated one is out of scope.

[Grab one from npm](https://www.npmjs.com/search?q=telegraf%20command) or [implement your own](https://github.com/thedevs-network/the-guard-bot/blob/develop/utils/cmd/parse.ts).

**Reply by tassoman** [reply]
Thanks for the complete reply, in the mean time I wrote `bot.hears(/mycommand\s?(.*)/i, (ctx) => {})`, then checking `ctx.match`.
Works with text message and commands either, but I feel too much inclusive.
My npm pulled telegraf v3.38.0.
So now I'm going to go with a middleware npm module, I'm not in typescript yet. 🔰

**Reply by wojpawlik** [reply]
> Works with text message and commands either, but I feel too much inclusive.

https://github.com/thedevs-network/the-guard-bot/blob/8b9babf987da1b48845be6ba809577abe50664a5/utils/cmd/parse.ts#L38-L46 (it's for 3.38.0 and needs to be updated for 4.x)

**Comment by Loskir** [comment]
I guess the easiest solution is

```js
bot.hears(/^\/auth (\w+) (.+)$/, (ctx) => {
  // ctx.match[1] - username
  // ctx.match[2] - password
})
```

**Reply by wojpawlik** [reply]
This will match monospaced "commands" as well. Related: #898

**Reply by Loskir** [reply]
Still valid in 99.9% use cases. Also easy for newbies
